invenio_records_resources/references/entity_resolvers/results.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 Graz University of Technology.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Resolver for result items."""
+
+from invenio_access.permissions import system_identity
+
+from .base import EntityProxy, EntityResolver
+
+
+class ResultItemProxy(EntityProxy):
+    """Resolver proxy for a result item entity."""
+
+    def __init__(self, resolver, ref_dict, service):
+        """Constructor."""
+        super().__init__(resolver, ref_dict)
+        self.service = service
+
+    def _resolve(self):
+        """Resolve the result item from the proxy's reference dict."""
+        id_ = self._parse_ref_dict_id()
+        return self.service.read(system_identity, id_).to_dict()
+
+    def get_needs(self, ctx=None):
+        """Return needs."""
+        # TODO: service.config.permission_policy_cls.can_read or []
+        #       Since resolving is done by the system, might not be applicable
+        return []
```

Comment:
comment: indeed in this case it's not needed since the `service.read` already applies the permission check (see also my comment on the configurable identity though)